turbopack/crates/turbopack-core/src/resolve/options.rs

```diff
@@ -94,7 +94,17 @@ pub enum ResolveInPackage {
 #[turbo_tasks::value(shared)]
 #[derive(Clone)]
 pub enum ImportMapping {
-    External(Option<RcStr>, ExternalType),
+    /// If specified, the optional name overrides the request, importing that external instead.
+    External(Option<RcStr>, ExternalType, ExternalTraced),
+    /// Only make the request external if the request is resolvable from the specified directory.
+    ///
+    /// If specified, the optional name overrides the request, importing that external instead.
+    PrimaryAlternativeExternal(
+        Option<RcStr>,
+        ExternalType,
+        ExternalTraced,
+        ResolvedVc<FileSystemPath>,
+    ),
```

Comment:
Could we use named fields for that to make it more clear what each argument means?